Equipment Details - First Draft

rev-development opened the pull request.

rev-development committed:
First Draft

scott181182 commented on frontend/components/metric-progress-circular.vue:
```diff
@@ -0,0 +1,29 @@
+
+
+interface Props {
+  equipment: MockEquipment;
```
You don't need a whole equipment here. Instead take a `Metric` (NOT a `MetricKey`)

scott181182 commented on frontend/components/metric-progress-circular.vue:
```diff
@@ -0,0 +1,29 @@
+
+// Unwrapped in template
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const props = defineProps<Props>();
```
Might as well unwrap it here: `const { metric } = defineProps<Props>();`

scott181182 commented on frontend/components/metric-select-card.vue:
```diff
@@ -0,0 +1,29 @@
+}
+
+// Unwrapped in template
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
There's a reason this is a default rule. Just unwrap it here.

scott181182 commented on frontend/components/equipment-details.vue:
```diff
@@ -0,0 +1,75 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const props = defineProps<Props>();
+
+const activeMetric: Ref<MetricKey> = ref("oee");
```
What's the purpose of `activeMetric`? Does it just highlight a metric when you click on it? Would this imply some extra interactivity that isn't there?

rev-development replied:
activeMetric determines which subcomponents are currently displayed

scott181182 commented on frontend/pages/index.vue:
```diff
@@ -1,23 +1,50 @@
-          :key="equipment.id"
-          :equipment="equipment"
-        />
+      <v-col v-if="activeEquipment" cols="10" class="border-e-md">
```
I think we discussed having a separate **page** for the equipment details. I suggest reverting this dashboard page back to where it was, and making a separate page with the equipment details. I reiterate this since there shouldn't be any value to the customer to look at other equipment while they're looking at the details for another.

scott181182 commented on frontend/utils/equipment.ts:
```diff
@@ -0,0 +1,44 @@
+export class MockEquipment {
+  data: IMockEquipment;
+
+  constructor(equipment: IMockEquipment) {
```
Passing in an object to a constructor is a bit of an anti-pattern. I'd recommend returning to the previous structure of mock equipment to avoid over-engineering, but I won't press that in this PR.

rev-development committed:
Changes in response to PR #13

Equipment Details made into separate pages (note: the routing is heavily cheesed, this will be corrected with graphql connection)
Components using MetricKey and Equipment props have been simplified to using a single Metric prop

scott181182 committed:
revert(frontend): changed dashboard layout back to grid to retiain constant-height cards

scott181182 committed:
refactor(frontend): unwrapped active prop instead of using it implicitly

scott181182 committed:
refactor(frontend): reorganized mocks and added back some metric computation that will help with GraphQL integration later

scott181182 committed:
feat(frontend): added back button to equipment detail page

scott181182 reviewed (approved):
I've patched some remaining stuff up prior to the standup this morning. I think this PR is good, then we can talk about what the end-state of the detail page is
